src/test/java/com/google/sps/DeleteUserServletTest.java

```diff
@@ -0,0 +1,105 @@
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.Gson;
+import com.google.sps.data.LoggedOutException;
+import com.google.sps.data.Player;
+import com.google.sps.data.PlayerDatabase;
+import com.google.sps.servlets.DeleteUserServlet;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** Tests that the DeleteUserServlet servlet correctly deletes a player from the database */
+@RunWith(JUnit4.class)
+public final class DeleteUserServletTest {
+  private static final String AUTH_DOMAIN = "email.com";
+  private static final String CURR_USER_ID = "testid";
+  private static final String USER_ID_KEY_PATH =
+      "com.google.appengine.api.users.UserService.user_id_key";
+  private static Map<String, Object> USER_ID_CONFIG =
+      new HashMap<String, Object>() {
+        {
+          put(USER_ID_KEY_PATH, CURR_USER_ID);
+        }
+      };
+  // creating a test user service
+  private final LocalUserServiceTestConfig localUserServiceTestConfig =
+      new LocalUserServiceTestConfig().setOAuthUserId(CURR_USER_ID);
+  private final LocalDatastoreServiceTestConfig localDatastoreServiceTestConfig =
+      new LocalDatastoreServiceTestConfig();
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(localUserServiceTestConfig, localDatastoreServiceTestConfig)
+          .setEnvIsAdmin(false)
+          .setEnvAuthDomain(AUTH_DOMAIN)
+          .setEnvEmail(EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAttributes(USER_ID_CONFIG);
+  private DatastoreService localDatastore;
+  private UserService localUserService;
+  private PlayerDatabase playerDatabase;
+  private DeleteUserServlet deleteUserServlet;
+  private static final String LOGGED_OUT_EXCEPTION =
+      "Player is currently logged out. Cannot process null user.";
+  private static final String DISPLAY_NAME = "TestName";
+  private static final String EMAIL = "TestName@email.com";
+  private static final Gson gson = new Gson();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    // initialize local user + datastore service.
+    helper.setUp();
+    this.localUserService = UserServiceFactory.getUserService();
+    this.localDatastore = DatastoreServiceFactory.getDatastoreService();
+    this.deleteUserServlet = new DeleteUserServlet();
+    this.deleteUserServlet.init();
+    this.playerDatabase = new PlayerDatabase(this.localDatastore, this.localUserService);
+    MockitoAnnotations.initMocks(this);
```

Comment:
small nit, but I would but the this.playerDatabase initialization above the this.deleteUserServlet initialization because that feels more logically consistent.

Reply:
It throws a null pointer exception otherwise